falcon/cluster/cluster.py

```diff
@@ -1,45 +1,51 @@
 import gc
 import logging
 import math
-import pickle
 import tempfile
-from typing import Callable, Iterator, List, Optional, Tuple
+from multiprocessing.pool import ThreadPool
+from typing import Iterator, List, Optional, Tuple
 
 import faiss
 import joblib
+import lance
+import more_itertools as mit
```

Comment:
**todo:** This either needs to be added to the dependencies or replaced by built-in functionality. What you're using is straightforward functionality, so I'd lean towards avoiding an additional dependency.